File: financial-analysis/skills/autoresearch-info-index/scripts/agent_reach_deploy_check.py
#!/usr/bin/env python3
"""
Run a local deployment check for a separate Agent Reach installation.
"""

from __future__ import annotations

import argparse
import json
import sys
from pathlib import Path

SCRIPT_DIR = Path(__file__).resolve().parent
if str(SCRIPT_DIR) not in sys.path:
    sys.path.insert(0, str(SCRIPT_DIR))

from agent_reach_deploy_check_runtime import load_json, run_agent_reach_deploy_check, write_json


def emit_json(payload: dict, *, stream: object = sys.stdout) -> None:
    text = json.dumps(payload, indent=2, ensure_ascii=False)
    try:
        print(text, file=stream)
    except UnicodeEncodeError:
        print(json.dumps(payload, indent=2, ensure_ascii=True), file=stream)


def parse_args() -> argparse.Namespace:
    parser = argparse.ArgumentParser(
        description="Check whether a separate Agent Reach deployment is present, pinned, and usable."
    )
    parser.add_argument("input", nargs="?", help="Optional path to a deploy-check request JSON file")
    parser.add_argument("--install-root", help="Override the Agent Reach install root to inspect")
    parser.add_argument("--pseudo-home", help="Override the Agent Reach pseudo-home to inspect")
    parser.add_argument("--python-binary", help="Override the expected full Python binary path")
    parser.add_argument("--output", help="Optional path to save the result JSON")
    parser.add_argument("--markdown-output", help="Optional path to save the markdown report")
    parser.add_argument(
        "--quiet",
        action="store_true",
        help="Suppress stdout JSON output and only write requested files.",
    )
    return parser.parse_args()


def main() -> None:
    args = parse_args()
    try:
        payload = load_json(Path(args.input).resolve()) if args.input else {}
        if args.install_root:
            payload["install_root"] = args.install_root
        if args.pseudo_home:
            payload["pseudo_home"] = args.pseudo_home
        if args.python_binary:
            payload["python_binary"] = args.python_binary
        result = run_agent_reach_deploy_check(payload)
        if not args.quiet:
            emit_json(result)

        if args.output:
            write_json(Path(args.output).resolve(), result)
        if args.markdown_output:
            output_path = Path(args.markdown_output).resolve()
            output_path.parent.mkdir(parents=True, exist_ok=True)
            output_path.write_text(result.get("report_markdown", ""), encoding="utf-8")
        sys.exit(0 if result.get("core_channels_ready") else 1)
    except Exception as exc:
        emit_json({"status": "ERROR", "message": str(exc)}, stream=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    main()

File: apps/financial-research-cli/src/workflows/agentReachDeployCheck.mjs
import { pushOutputArgs, requireInput } from "./workflowUtils.mjs";

export const agentReachDeployCheckCommand = {
  name: "agent-reach-deploy-check",
  script: "agent_reach_deploy_check.py",
  wrapper: "run_agent_reach_deploy_check.cmd",
  description: "Check local Agent Reach bridge prerequisites.",
  validateOptions(options) {
    return requireInput(this.name, options);
  },
  buildArgs(options) {
    const args = [options.input];
    if (options.installRoot) args.push("--install-root", options.installRoot);
    if (options.pseudoHome) args.push("--pseudo-home", options.pseudoHome);
    if (options.pythonBinary) args.push("--python-binary", options.pythonBinary);
    return pushOutputArgs(args, options);
  },
  buildSummary(payload) {
    const binaries = payload.binaries || {};
    const channels = payload.channels || {};
    return {
      deploy_status: payload.status || "",
      core_channels_ready: Boolean(payload.core_channels_ready),
      channel_ok_count: Object.values(channels).filter((value) => value === "ok" || value === true).length,
      missing_binary_count: Object.values(binaries).filter((value) => value !== "ok" && value !== true).length,
    };
  },
  renderSummary(contract) {
    return `Command: agent-reach-deploy-check\nStatus: ${contract.status}\nWorkflow: ${contract.workflow_kind}`;
  },
};
